Zend/zend_execute.c

```diff
@@ -4459,7 +4459,24 @@ ZEND_API void zend_cleanup_unfinished_execution(zend_execute_data *execute_data,
 	cleanup_live_vars(execute_data, op_num, catch_op_num);
 }
 
-ZEND_API HashTable *zend_unfinished_execution_gc(zend_execute_data *execute_data, zend_execute_data *call, zend_get_gc_buffer *gc_buffer)
+ZEND_API ZEND_ATTRIBUTE_DEPRECATED HashTable *zend_unfinished_execution_gc(zend_execute_data *execute_data, zend_execute_data *call, zend_get_gc_buffer *gc_buffer)
+{
+	bool suspended_by_yield = false;
+
+	if (Z_TYPE_INFO(EX(This)) & ZEND_CALL_GENERATOR) {
+		ZEND_ASSERT(EX(return_value));
+
+		/* The generator object is stored in EX(return_value) */
+		zend_generator *generator = (zend_generator*) EX(return_value);
+		ZEND_ASSERT(execute_data == generator->execute_data);
+
+		suspended_by_yield = !(generator->flags & ZEND_GENERATOR_CURRENTLY_RUNNING);
+	}
+
+	return zend_unfinished_execution_gc_ex(execute_data, call, gc_buffer, suspended_by_yield);
+}
+
+ZEND_API HashTable *zend_unfinished_execution_gc_ex(zend_execute_data *execute_data, zend_execute_data *call, zend_get_gc_buffer *gc_buffer, bool suspended_by_yield)
```

Comment:
This avoids breaking ABI, although zend_unfinished_execution_gc was only introduced in the last _patch_ release.

Reply:
Did we increase the ABI number? As I don't know how ABI breaks are meant to be handled for stuff released after GA. @cmb69 or @derickr do you know what the policy is?

Reply:
The ABI number is the last time ABI was broken (which defacto is every minor version). Here ABI is not broken, but the original function preserved and a new one added. Hence it's a non-issue.

Reply:
> As I don't know how ABI breaks are meant to be handled for stuff released after GA.

Not at all; keeping ABI compat after first GA of a minor is a pretty strict policy (otherwise all extensions etc. would need to be rebuilt).

But yeah, not an issue here.